### ISSUE TITLE
Many tiny nitpicks 2017-12-27

### ISSUE THREAD

**Issue by joncampbell123** [joncampbell123/dosbox-x#424]
Many tiny nitpicks coming up next, each one too small to make an issue for them:

- src/hardware/pic.cpp: declare "cascade IRQ" variable and match to that instead of per-function if() statements to enforce the same. It would initialize it to 2 for IBM PC/AT mode and 7 for PC-98 mode.

**Comment by roytam1** [comment]
what about keyboard in PC-98 mode?

**Comment by joncampbell123** [comment]
@roytam1 Keyboard emulation is already there, except the commands TO the keyboard to control caps/kana state and such, at the hardware level and BIOS level. What's incomplete is the DOS console driver handling of it.

**Comment by joncampbell123** [comment]
@roytam1 Since this conversation, I have expanded the BIOS and DOS CON driver support in PC-98 to include function keys, arrow keys, etc. Most of the DOSKEY-like editing functions in the IBM PC mode of the DOSBox shell now exist in PC-98 mode as well.

**Comment by roytam1** [comment]
@joncampbell123 sounds great.
But from my testing build with both vs2013/2015, it seems that keyboard mapping is still broken(only 1,q,a,arrow keys and numpad enter key are working).

dosbox.conf has 2 lines only for testing in this situation:
````
[dosbox]
machine=pc98
````

**Comment by joncampbell123** [comment]
The mapper UI in the latest build offers a way to tell if any keys are triggering. Use CTRL+F1 to bring up the mapper UI. When you press keys, the key that DOSBox-X sends to the guest should light up.

**Comment by roytam1** [comment]
In key mapper, every key works.
but in PC-98 dosbox prompt, they don't work(except the keys mentioned earlier), in non-PC98 mode every key works as well.
![keymap](https://user-images.githubusercontent.com/1192577/34476236-e9a4f5c4-efc9-11e7-98cf-5408a9d69520.png)

**Comment by joncampbell123** [comment]
There is a known issue with the mapper UI where entering the mapper through the keyboard shortcut leaves the CTRL and SHIFT keys "stuck" in the guest. Does tapping the CTRL and SHIFT keys help?

Second, the mapper UI allows you to change and redefine the keys. The default bindings should show bindings for all the standard keys on the keyboard.

**Comment by roytam1** [comment]
> Does tapping the CTRL and SHIFT keys help?

nope, still 3 keys of alphanumeric area working in PC98 mode, all keys are working in non-PC98 mode.

**Comment by joncampbell123** [comment]
@roytam1 Recent builds of DOSBox-X removed the hack code in SDL 1.x that loads the US keyboard layout for DOSBox-X itself. See if that helps with the keyboard issue.
